packages/core/contracts/oracle/implementation/StakerSnapshot.sol

```diff
@@ -0,0 +1,244 @@
+// SPDX-License-Identifier: AGPL-3.0-only
```

Comment:
this is the primary new contract implemented in this PR. it is meant to be inherited from VotingV2.sol and provides the ability to Stake and snapshot staked amounts in the voting contract. it is based off in part the [accelerating distributor](https://github.com/across-protocol/across-token/blob/master/contracts/AcceleratingDistributor.sol) from across v2 and the [ERC20Snapshot](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/extensions/ERC20Snapshot.sol) from openzepplin.